src/lib/addons/slack-app.ts

```diff
@@ -17,64 +21,103 @@ interface ISlackAppAddonParameters {
 export default class SlackAppAddon extends Addon {
     private msgFormatter: FeatureEventFormatter;
 
+    private accessToken?: string;
+
     private slackClient?: WebClient;
 
+    private slackChannels?: ConversationsListResponse['channels'];
+
+    private slackChannelsCacheTimeout?: NodeJS.Timeout;
+
     constructor(args: IAddonConfig) {
         super(slackAppDefinition, args);
         this.msgFormatter = new FeatureEventFormatterMd(
             args.unleashUrl,
             LinkStyle.SLACK,
         );
+        this.startCacheInvalidation();
     }
 
     async handleEvent(
         event: IEvent,
         parameters: ISlackAppAddonParameters,
     ): Promise<void> {
-        const { accessToken } = parameters;
+        try {
+            const { accessToken } = parameters;
 
-        if (!accessToken) return;
+            if (!accessToken) return;
 
-        if (!this.slackClient) {
-            this.slackClient = new WebClient(accessToken);
-        }
+            if (!this.slackClient || this.accessToken !== accessToken) {
+                this.slackClient = new WebClient(accessToken);
+            }
```

Comment:
Oops, forgot about assigning this 🙈 Addressed in https://github.com/Unleash/unleash/pull/4284/commits/b4fa168f4005e6a0d504d6dc60fab4f3b4da9e0c

This check is done in order to initialize a new `WebClient` with the new token in case we change it in the meantime.